lib/game_play.rb

```diff
@@ -59,25 +60,37 @@ def create_players(selection)
     end
   end
 
+  def get_user_selection_between(min, max)
+    value = get_user_selection
+    if !value.between?(min, max)
+      @ui.between(min, max)
+      value = get_user_selection_between(min, max)
```

Comment:
The key thing here is that there is no check after the recursive call.
You might have wrapped the recursive call in a loop that checked the result of the recursive call?